airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -46,6 +48,23 @@ def __init__(
         self.schema: Mapping[str, Any] = schema  # type: ignore[assignment]
         self.sobject_options = sobject_options
 
+    def decode(self, chunk):
+        """
+        Most Salesforce instances use UTF-8, but some use ISO-8859-1.
+        By default, we'll decode using UTF-8, and fallback to ISO-8859-1 if it doesn't work.
+        See implementation considerations for more details https://developer.salesforce.com/docs/atlas.en-us.api.meta/api/implementation_considerations.htm
+        """
+        if self.encoding == DEFAULT_ENCODING:
+            try:
+                decoded = chunk.decode(self.encoding)
+                return decoded
+            except UnicodeDecodeError as e:
+                self.encoding = "ISO-8859-1"
+                self.logger.info(f"Could not decode chunk. Falling back to {self.encoding} encoding. Error: {e}")
```

Comment:
If all the messages are encoded with `ISO-8859-1`, the log will be emitted for every message. So maybe remove this log or only log it once?

Reply:
@tuliren this will only run once because of [the if statement on ](https://github.com/airbytehq/airbyte/pull/12576/files#diff-0572edeedd561901f8110c0f88b48efde87d1ed91247c0eb189c7229da12090dR57)